text/0000-api-doc-frontpage-styleguide.md

```diff
@@ -0,0 +1,174 @@
+of a crate. These are the guidelines we'll be using for the rust-lang crates, but are also meant to
+help guide community crate authors.
+
+# Motivation
+
+A user's first experience with a crate is the front page of the crate's documentation. Taken
+together with the API documentation, the front page is a key piece of understanding a crate's
+purpose, how to use it, and its limitations.  Making sure this front page is clear will help users
+of your crate get the most out of it.
+
+# Detailed design
+
+The front page of the crate should cover four main points.  In this RFC, we cover each in turn.
+
+* [Introduction text to the crate](#introduction-text-to-the-crate) 
+* [First Example](#first-example) 
+* [Crate Capabilities](#crate-capabilities) 
+* [Source Examples](#source-examples)
+
+## Introduction text to the crate
+
+The first thing potential users of your crates will see is the introductory summary. This section is
+a good place to introduce the what and the why the crate. A good introduction is concise but also
+provides enough information for the reader to understand the purpose of the crate. In addition, a
+good introduction provides a learning path for the user as they read the rest of the documentation.
+
+Here's an example of a good introduction from the [log
+crate](https://doc.rust-lang.org/log/log/index.html):
+    
+> A lightweight logging facade.  
+>
+> A logging facade provides a single logging API that abstracts over the actual logging
+> implementation. Libraries can use the logging API provided by this crate, and the consumer
+> of those libraries can choose the logging framework that is most suitable for its use case.
+>
+> If no logging implementation is selected, the facade falls back to a "noop" implementation
+> that ignores all log messages. The overhead in this case is very small - just an integer load,
+> comparison and jump.  > > A log request consists of a target, a level, and a body. A target
+> is a string which defaults to the module path of the location of the log request, though that
+> default may be overridden. Logger implementations typically use the target to filter requests
+> based on some user configuration.
+
+There are several things that make this a good introduction section:
+    
+* A concise explanation of the crate's purpose.
+* The summary describes how the crate behaves in 3 short paragraphs.
+* Even if you don't want to read the rest of the `README`, you have enough to get started.
+* Readable by a layman, no jargon is introduced in the first few paragraphs.
+    
+What do we mean by "no jargon is introduced in the first few paragraphs"? Using the `rand` crate as
+an example - the crate's initial documentation probably should _not_ include a discussion of
+cryptographically secure random number generators since its primary purpose, for most Rust
+developers, will be to produce any random number. This information should instead be provided in a
```

Comment:
> primary purpose, for most Rust developers, will be to produce any random number

That is a great misconception. It is very easy to incorrectly decide on a random number generator for pretty much all the use cases without taking in account the cryptosafety of the generator and the need for it. Even the trivial die-rolling-games are flawed without a crypto-safe generator in some circumstances.

That being said, I do agree that discussion about generator’s crypto-safety should be elsewhere (such as the generators themselves)